Sources/DistributedActors/Cluster/ClusterControl.swift

```diff
@@ -145,4 +155,77 @@ public struct ClusterControl {
+            for node in nodes {
+                group.addTask {
+                    try await self.awaitMemberStatus(is: status, within: within, node: node)
+                }
+                try await group.next() // this is required to propagate error and cancel all tasks
+            }
+        }
+    }
+
+    public func ensureNodes(atLeast atLeastStatus: Cluster.MemberStatus, within: Duration, nodes: [UniqueNode]) async throws {
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            for node in nodes {
+                group.addTask {
+                    try await self.awaitMemberStatus(atLeast: atLeastStatus, within: within, node: node)
+                }
+                try await group.next() // this is required to propagate error and cancel all tasks
+            }
+        }
+    }
+
+    public func awaitMemberStatus(atLeast atLeastStatus: Cluster.MemberStatus, within: Duration, node: UniqueNode) async throws {
+        try await self.awaitMemberStatusEventually(within: within) { membership in
+            guard let foundMember = membership.uniqueMember(node) else {
+                if atLeastStatus == .down || atLeastStatus == .removed {
+                    // so we're seeing an already removed member, this can indeed happen and is okey
+                    return
+                } else {
+                    throw Cluster.MembershipError.notFound(node)
+                }
+            }
+
+            if atLeastStatus <= foundMember.status {
+                throw Cluster.MembershipError.atLeastStatusRequirementNotMet(expectedAtLeast: atLeastStatus, actual: foundMember.status)
+            }
+        }
+    }
+
+    public func awaitMemberStatus(is status: Cluster.MemberStatus, within: Duration, node: UniqueNode) async throws {
+        try await self.awaitMemberStatusEventually(within: within) { membership in
+            guard let foundMember = membership.uniqueMember(node) else {
+                throw Cluster.MembershipError.notFound(node)
+            }
+
+            if status != foundMember.status {
+                throw Cluster.MembershipError.statusRequirementNotMet(expected: status, actual: foundMember.status)
+            }
+        }
+    }
+
+    private func awaitMemberStatusEventually(within: Duration, interval: Duration = .milliseconds(100), _ block: (Cluster.Membership) async throws -> Void) async throws {
+        let deadline = ContinuousClock.Instant.fromNow(within)
+
+        var lastError: Error?
+        while deadline.hasTimeLeft() {
+            let membership = await self.membershipSnapshot
+            do {
+                try await block(membership)
+                return
+            } catch {
+                lastError = error
+                try await Task.sleep(nanoseconds: UInt64(interval.nanoseconds))
+            }
+        }
+
+        throw Cluster.MembershipError.awaitStatusTimedOut(within, lastError)
+    }
 }
```

Comment:
@ktoso Are these APIs what you have in mind?

Reply:
Yes those look mostly good, commented a little bit on the shapes; I guess the testkit ones may have been a bit inconsistent?